core/base-service/legacy-request-handler.js

```diff
@@ -1,143 +0,0 @@
-// parameter not working at all (which is undesirable, but easy to debug)
-// rather than indeterminate behavior that depends on the cache state
-// (undesirable and hard to debug).
-//
-// Pass just the handler function as shorthand.
-function handleRequest(cacheHeaderConfig, handlerOptions) {
-  if (!cacheHeaderConfig) {
-    throw Error('cacheHeaderConfig is required')
-  }
-
-  if (typeof handlerOptions === 'function') {
-    handlerOptions = { handler: handlerOptions }
-  }
-
-  const allowedKeys = flattenQueryParams(handlerOptions.queryParams)
-  const { cacheLength: serviceDefaultCacheLengthSeconds } = handlerOptions
-
-  return (queryParams, match, end, ask) => {
-    /*
-    This is here for legacy reasons. The badge server and frontend used to live
-    on two different servers. When we merged them there was a conflict so we
-    did this to avoid moving the endpoint docs to another URL.
-
-    Never ever do this again.
-    */
-    if (match[0] === '/endpoint' && Object.keys(queryParams).length === 0) {
-      ask.res.statusCode = 301
-      ask.res.setHeader('Location', '/endpoint/')
-      ask.res.end()
-      return
-    }
-
-    // `defaultCacheLengthSeconds` can be overridden by
-    // `serviceDefaultCacheLengthSeconds` (either by category or on a badge-
-    // by-badge basis). Then in turn that can be overridden by
-    // `serviceOverrideCacheLengthSeconds` (which we expect to be used only in
-    // the dynamic badge) but only if `serviceOverrideCacheLengthSeconds` is
-    // longer than `serviceDefaultCacheLengthSeconds` and then the `cacheSeconds`
-    // query param can also override both of those but again only if `cacheSeconds`
-    // is longer.
-    //
-    // When the legacy services have been rewritten, all the code in here
-    // will go away, which should achieve this goal in a simpler way.
-    //
-    // Ref: https://github.com/badges/shields/pull/2755
-    function setCacheHeadersOnResponse(res, serviceOverrideCacheLengthSeconds) {
-      setCacheHeaders({
-        cacheHeaderConfig,
-        serviceDefaultCacheLengthSeconds,
-        serviceOverrideCacheLengthSeconds,
-        queryParams,
-        res,
-      })
-    }
-
-    const filteredQueryParams = {}
-    allowedKeys.forEach(key => {
-      filteredQueryParams[key] = queryParams[key]
-    })
-
-    // In case our vendor servers are unresponsive.
-    let serverUnresponsive = false
-    const serverResponsive = setTimeout(() => {
-      serverUnresponsive = true
-      ask.res.setHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
-      const badgeData = coalesceBadge(
-        filteredQueryParams,
-        { label: 'vendor', message: 'unresponsive' },
-        {}
-      )
-      const svg = makeBadge(badgeData)
-      const extension = (match.slice(-1)[0] || '.svg').replace(/^\./, '')
-      setCacheHeadersOnResponse(ask.res)
-      makeSend(extension, ask.res, end)(svg)
-    }, 25000)
```

Comment:
I guess we've just totally dropped this in favour of express `this.server.setTimeout()` ?